nautobot/extras/models/models.py

```diff
@@ -644,6 +644,48 @@ def size(self):
             return None
 
 
+#
+# Notes
+#
+
+
+@extras_features("graphql", "webhooks")
+class Notes(BaseModel, ChangeLoggedModel):
+    """
+    Notes allows anyone with proper permissions to add a note to an object.
+    """
+
+    assigned_object_type = models.ForeignKey(to=ContentType, on_delete=models.CASCADE)
+    assigned_object_id = models.UUIDField(db_index=True)
+    assigned_object = GenericForeignKey(ct_field="assigned_object_type", fk_field="assigned_object_id")
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        related_name="note",
+        blank=True,
+        null=True,
+    )
+    user_name = models.CharField(max_length=150, editable=False)
```

Comment:
What's the purpose of the `user_name` field? Is it just to have a record in the case where the `User` gets deleted, or does it do something more than that? Should we just use `user.username` or `user.first_name`/`user.last_name` on the fly as appropriate instead?